app/src/main/java/org/p2p/wallet/swap/ui/jupiter/settings/presenter/SwapContentSettingsMapper.kt

```diff
@@ -0,0 +1,242 @@
+    ): List<AnyCellItem> = mutableListOf<AnyCellItem>().apply {
+        addRouteCell(routes, activeRoute, jupiterTokens)
+        this += commonMapper.getNetworkFeeCell()
+        addAccountFeeCell()
+        addLiquidityFeeCell(routes, activeRoute, jupiterTokens)
+        addEstimatedFeeCell()
+        addMinimumReceivedCell(slippage, tokenBAmount, tokenB)
+        this += commonMapper.createHeader(R.string.swap_settings_slippage_title)
+        addAll(commonMapper.getSlippageList(slippage))
+    }
+
+    private fun MutableList<AnyCellItem>.addRouteCell(
+        routes: List<JupiterSwapRoute>,
+        activeRoute: Int,
+        jupiterTokens: List<JupiterSwapToken>,
+    ) {
+        val isBestRoute = activeRoute == SwapStateManager.DEFAULT_ACTIVE_ROUTE_ORDINAL
+        val route = routes.getOrNull(activeRoute)
+        this += FinanceBlockCellModel(
+            leftSideCellModel = LeftSideCellModel.IconWithText(
+                firstLineText = TextViewCellModel.Raw(
+                    text = TextContainer(R.string.swap_settings_route_title),
+                ),
+                secondLineText = TextViewCellModel.Raw(
+                    text = TextContainer(formatRouteString(route, jupiterTokens))
+                ),
+            ),
+            rightSideCellModel = RightSideCellModel.SingleTextTwoIcon(
+                text = if (isBestRoute) {
+                    TextViewCellModel.Raw(text = TextContainer(R.string.swap_settings_route_best))
+                } else {
+                    null
+                },
+                firstIcon = ImageViewCellModel(
+                    icon = DrawableContainer(R.drawable.ic_chevron_right),
+                    iconTint = R.color.icons_mountain,
+                )
+            ),
+            payload = SwapSettingsPayload.ROUTE,
+            styleType = FinanceBlockStyle.BASE_CELL,
+        )
+    }
+
+    private fun formatRouteString(route: JupiterSwapRoute?, jupiterTokens: List<JupiterSwapToken>): String {
+        if (route == null) return ""
+        var result = ""
+        route.marketInfos.forEachIndexed { index, marketInfo ->
+            result = if (index != route.marketInfos.lastIndex) {
+                result.plus(findTokenSymbolByMint(marketInfo.inputMint, jupiterTokens)).plus("→")
+            } else {
+                result.plus(findTokenSymbolByMint(marketInfo.outputMint, jupiterTokens))
+            }
+        }
+        return result
+    }
+
+    private fun MutableList<AnyCellItem>.addMinimumReceivedCell(
+        slippage: Double,
+        tokenBAmount: BigDecimal?,
+        tokenB: SwapTokenModel
+    ) {
+        val secondLineText = if (tokenBAmount == null) {
+            TextViewCellModel.Skeleton(
+                skeleton = leftSubtitleSkeleton()
+            )
+        } else {
+            TextViewCellModel.Raw(
+                text = TextContainer(tokenBAmount.multiply(slippage.toBigDecimal()).formatToken(tokenB.decimals))
```

Comment:
extract tokenBAmount.multiply(slippage.toBigDecimal()).formatToken(tokenB.decimals) to variable to split meaning